src/Autofac/PipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,135 @@
+﻿// This software is part of the Autofac IoC container
+// Copyright © 2020 Autofac Contributors
+// https://autofac.org
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using Autofac.Core.Resolving.Middleware;
+using Autofac.Core.Resolving.Pipeline;
+
+namespace Autofac
+{
+    /// <summary>
+    /// Extension methods for adding pipeline middleware.
+    /// </summary>
+    public static class PipelineBuilderExtensions
+    {
+        private const string AnonymousName = "unnamed";
```

Comment:
I just noticed this - `AnonymousName`. It seems like it's more about _describing_ the middleware, like for an exception message, than it is about _naming_ it, where 'name' has a connotation in the codebase like "this is a named/keyed component." If it's simply a describing string, it'd be good to keep "description" in the method parameter and variable names to avoid reuse of the term.

Reply:
I gather 'name' was chosen to ensure 'description' didn't yield someone putting full sentences in there and shooting themselves in the foot. Maybe 'descriptor' rather than 'description' would be a compromise.